utils/encryption.py

```diff
@@ -0,0 +1,100 @@
+import getpass
+import json
+import os
+from base64 import b64encode, b64decode
+from Crypto.Cipher import ChaCha20_Poly1305
+from Crypto.Random import get_random_bytes
+from Crypto.Protocol.KDF import scrypt
+
+from utils.logger import log
+
+
+def encrypt(pt, password):
+    salt = get_random_bytes(32)
+    key = scrypt(password, salt, key_len=32, N=2 ** 20, r=8, p=1)
+    nonce = get_random_bytes(12)
+    cipher = ChaCha20_Poly1305.new(key=key, nonce=nonce)
+    ct, tag = cipher.encrypt_and_digest(pt)
+    json_k = ["nonce", "salt", "ct", "tag"]
+    json_v = [b64encode(x).decode("utf-8") for x in (nonce, salt, ct, tag)]
+    result = json.dumps(dict(zip(json_k, json_v)))
+
+    return result
+
+
+def decrypt(ct, password):
+    try:
+        b64Ct = json.loads(ct)
+        json_k = ["nonce", "salt", "ct", "tag"]
+        json_v = {k: b64decode(b64Ct[k]) for k in json_k}
+
+        key = scrypt(password, json_v["salt"], key_len=32, N=2 ** 20, r=8, p=1)
+        cipher = ChaCha20_Poly1305.new(key=key, nonce=json_v["nonce"])
+        ptData = cipher.decrypt_and_verify(json_v["ct"], json_v["tag"])
+
+        return ptData
+    except (KeyError, ValueError):
+        print("Incorrect Password.")
+        exit(0)
+
+
+def create_encrypted_config(data, file_path):
+    if isinstance(data, dict):
+        data = json.dumps(data)
+    payload = bytes(data, "utf-8")
+    log.info("Create a password for the credential file")
+    cpass = getpass.getpass(prompt="Credential file password: ")
```

Comment:
Not getting this prompt 
```
INFO: "2020-10-05 11:21:11,580 - Initializing notification handlers
INFO: "2020-10-05 11:21:11,581 - Enabled Handlers: ['Audio', 'Pavlok']
INFO: "2020-10-05 11:21:11,584 - Reading credentials from: amazon_config.json
INFO: "2020-10-05 11:21:11,584 - Your configuration file is unencrypted, it will now be encrypted.
INFO: "2020-10-05 11:21:11,584 - Create a password for the credential file
```

Win 10, running without emulated terminal. This does work with an emulated terminal though.

Reply:
Running with no previous config:
![](https://cdn.discordapp.com/attachments/757272601977684028/762757877567324190/unknown.png "Running with no previous config")

Running with previously encrypted config:
![](https://cdn.discordapp.com/attachments/757272601977684028/762757975294214144/unknown.png "Running with previously encrypted config")

Running from previously unencrypted config:
![](https://cdn.discordapp.com/attachments/757272601977684028/762790712219533312/unknown.png "Running from previously unencrypted config")